### ISSUE TITLE
type of wrappedComponentRef

### ISSUE THREAD

**Issue by DiamondYuan** [ant-design/ant-design#15742]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
现在 ts 使用 from 时候，ref 类型只能是 any

### What does the proposed API look like?
无

**Comment by DiamondYuan** [comment]
```react
import { Form, Input } from 'antd';
import { FormComponentProps } from 'antd/lib/form';
import React, { Component } from 'react';

interface UserFormProps extends FormComponentProps {
  name?: string;
}

export class UserForm extends Component<UserFormProps> {
  render() {
    const { form, name } = this.props;
    return (
      <Form>
        <Form.Item>
          {form.getFieldDecorator('name', {
            initialValue: name,
          })(<Input />)}
        </Form.Item>
      </Form>
    );
  }
}

export default Form.create<UserFormProps>()(UserForm);
```


```react
import Form, { UserForm } from './user';
import * as React from 'react';

export type Instance<T> = T extends { new (...args: any[]): infer U }
  ? U
  : never;

export default class Parent extends React.Component {
  private childrenRef?: Instance<typeof UserForm>;

  setChildrenRef = (element: Instance<typeof UserForm>) => {
    this.childrenRef = element;
  };

  public render() {
    return (
      <div>
        <div
          onClick={() => {
            if (this.childrenRef) {
              console.log(this.childrenRef.props.form);
            }
          }}
        >
          print childrenRef form
        </div>
        <Form wrappedComponentRef={this.setChildrenRef} />;
      </div>
    );
  }
}

```

**Comment by afc163** [comment]
直接来 PR？

**Comment by DiamondYuan** [comment]
> 直接来 PR？

加一个高级类型来还原？

```typescript
export type WrappedFormComponent<P extends RcBaseFormProps> = React.ComponentClass<FormComponentProps & Omit<P, keyof RcBaseFormProps>>;

```

**Comment by afc163** [comment]
Fixed in #16242
